jaq-core/src/lib.rs

```diff
@@ -402,3 +404,82 @@ const LOG: &[(&str, usize, RunPtr, UpdatePtr)] = &[(
     |_, cv| box_once(Ok(debug(cv.1))),
     |_, cv, f| f(debug(cv.1)),
 )];
+
+#[cfg(feature = "format")]
+fn stringify(v: Val) -> String {
```

Comment:
I would actually make this a member function of Val, because it is a functionality which will be used implicitly in the interpreter. That is, `"world" | "Hello, \(.)"` will use this function for interpolation.
However I would rename this function; what do you think about to_string_or_clone()? I know it's a bit longer than stringify, but at least it's a bit more descriptive and clarifies the difference with respect to to_string().

Reply:
As a testament to my inexperience with Rust library design, I really screwed this up w.r.t. using `Rc` and cloning everywhere. I tried the `Cow` route and the `AsRef` route, but failed to make it "nice" when calling the function and `Val::str`.

Anyway, feel free to rework it all to reduce unnecessary cloning and such.

Reply:
I don't see a lot of unnecessary cloning. Don't worry, I think you're doing a good job.